lib/device/fan.js

```diff
@@ -87,6 +98,73 @@ export default class {
     }
   }
 
+  async internalSwingUpdate(value) {
+    try {
+      const newValue = value ? 'SWING_ON' : 'SWING_OFF';
+      // Don't continue if the new value is the same as before
+      if (this.cacheSwing === value) {
+        return;
+      }
+
+      await this.platform.sendDeviceUpdate(this.accessory, {
+        cmd: 'stateFan',
+        value: fanState[newValue],
+      });
+
+      // Cache the new state and log if appropriate
+      this.cacheSwing = newValue;
+      if (this.enableLogging) {
+        this.log('[%s] %s [%s].', this.name, platformLang.curState, newValue);
+      }
+    } catch (err) {
+      // Catch any errors during the process
+      const eText = parseError(err);
+      this.log.warn('[%s] %s %s.', this.name, platformLang.devNotUpdated, eText);
+
+      // Throw a 'no response' error and set a timeout to revert this after 2 seconds
+      setTimeout(() => {
+        this.service.updateCharacteristic(this.hapChar.On, this.cacheState === 'on');
```

Comment:
Also here the characteristic needs to be changed to `this.hapChar.SwingMode` and the value I guess something like `this.cacheSwing === 'SWING_ON ? 1 : 0`